src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtTokenProvider.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.hmcts.reform.sandl.snlapi.security;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.UnsupportedJwtException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sandl.snlapi.security.model.UserPrincipal;
+
+import java.util.Date;
+
+@Component
+public class JwtTokenProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(JwtTokenProvider.class);
+
+    @Value("${management.security.jwtSecret}")
+    private String jwtSecret;
+
+    @Value("${management.security.jwtExpirationInMs}")
+    private int jwtExpirationInMs;
+
+    public String generateToken(Authentication authentication) {
+
+        UserPrincipal userPrincipal = (UserPrincipal) authentication.getPrincipal();
+
+        Date now = new Date();
```

Comment:
Not sure if it can cause any problems, but probably we shouldn't use Date as it's old Java API (not thread-safe etc.)